From 11f08c63d9cceae7ef422fc0aebc8b69ef104e37 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 19 Apr 2019 18:25:30 +0200
Subject: [PATCH 013/171] drivers/base/core: revisit and add another annotation
 for lock_device_hotplug_sysfs

Hence, handling also these warnings:

drivers/base/core.c:863:1: warning: mutex 'device_hotplug_lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
drivers/base/core.c:857:6: note: mutex acquired here
        if (mutex_trylock(&device_hotplug_lock))
            ^

drivers/base/core.c:1281:2: warning: releasing mutex 'device_hotplug_lock' that was not held [-Wthread-safety-analysis]
        unlock_device_hotplug();
        ^
---
 drivers/base/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/base/core.c b/drivers/base/core.c
index dd39f0bcab53..4bd00ffa2c80 100644
--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -852,7 +852,7 @@ void unlock_device_hotplug(void) __releases_mutex(device_hotplug_lock)
 	mutex_unlock(&device_hotplug_lock);
 }
 
-int lock_device_hotplug_sysfs(void)
+int lock_device_hotplug_sysfs(void) __try_acquires_mutex(0, device_hotplug_lock)
 {
 	if (mutex_trylock(&device_hotplug_lock))
 		return 0;
-- 
2.17.1

